projects/js-packages/charts/src/components/line-chart/line-chart-annotation-label-popover.tsx

```diff
@@ -0,0 +1,109 @@
+import { Gridicon } from '@automattic/jetpack-components';
+import clsx from 'clsx';
+import { useEffect, useId, useRef, useState } from 'react';
+import { isSafari } from '../shared/utils';
+import styles from './line-chart.module.scss';
+import type { ButtonWithPopover, PopoverElement, ToggleEvent } from '../../types';
+import type { FC } from 'react';
+
+export const POPOVER_BUTTON_SIZE = 44;
+
+interface LineChartAnnotationLabelWithPopoverProps {
+	title: string;
+	subtitle?: string;
+	renderLabel: FC< { title: string; subtitle?: string } >;
+	renderLabelPopover: FC< { title: string; subtitle?: string } >;
+}
+
+const LineChartAnnotationLabelWithPopover: FC< LineChartAnnotationLabelWithPopoverProps > = ( {
+	title,
+	subtitle,
+	renderLabel,
+	renderLabelPopover,
+} ) => {
+	const popoverId = useId();
+	const buttonRef = useRef< HTMLButtonElement >( null );
+	const popoverRef = useRef< HTMLDivElement >( null );
+	const [ isPositioned, setIsPositioned ] = useState( false );
+
+	useEffect( () => {
+		const button = buttonRef.current;
+		const popover = popoverRef.current;
+
+		if ( ! button || ! popover ) return;
+
+		const positionPopover = () => {
+			// Popover positioning in Safari is complicated due to issues with SVG foreign objects, so let it be positioned in the centre of the viewport.
+			if ( ! isSafari ) {
+				const buttonRect = button.getBoundingClientRect();
+				popover.style.left = `${ buttonRect.right }px`;
+				popover.style.top = `${ buttonRect.top }px`;
+			}
+
+			setIsPositioned( true );
+		};
+
+		// Position when popover shows
+		popover.addEventListener( 'toggle', ( e: ToggleEvent ) => {
+			if ( e.newState === 'open' ) {
+				positionPopover();
+			}
+		} );
+
+		// Initial positioning if already open
+		try {
+			if ( popover.matches( ':popover-open' ) ) {
+				positionPopover();
+			}
+		} catch {
+			// Ignore errors in test environments (e.g., JSDOM does not support :popover-open)
+		}
+	}, [] );
+
+	return (
+		<div className={ styles[ 'line-chart__annotation-label' ] }>
+			<button
+				ref={ buttonRef }
+				{ ...( { popovertarget: popoverId } as ButtonWithPopover ) }
```

Comment:
Using the popover API for these rather than a modal as it has:

Built-in Accessibility: The popover API automatically handles:

- Focus management
- ARIA attributes
- Keyboard navigation (Escape key)
- Screen reader announcements

Browser Behavior: Native handling of:

- Click outside to close
- Proper z-index stacking
- Scroll behavior
- Mobile touch interactions

Browser support is good: https://caniuse.com/mdn-api_htmlelement_popover